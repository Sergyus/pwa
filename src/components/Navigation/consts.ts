export const headerMenu: { name: string; link: string }[] = [
  {
    name: 'home',
    link: '/',
  },
  {
    name: 'blog',
    link: '/blog',
  },
  {
    name: 'about',
    link: '/about',
  },
  {
    name: 'Ui',
    link: '/ui',
  },
  {
    name: 'Pwa',
    link: '/pwa',
  },
  {
    name: 'Apollo',
    link: '/apollo',
  },
];
