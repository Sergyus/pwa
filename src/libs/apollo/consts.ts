export const CONFIG = {
  url: 'https://api.graphqlplaceholder.com/',
};
