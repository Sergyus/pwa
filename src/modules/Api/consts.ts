export const CONFIG = {
  url: 'https://jsonplaceholder.typicode.com/',
};
