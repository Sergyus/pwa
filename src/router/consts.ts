import loadable from '@loadable/component';
import { RouterConfigType } from './typings';
import PostService from '../modules/Post';
import App from '../app';

/**
 * Router Config
 */
export const config: RouterConfigType[] = [
  {
    component: App,
    routes: [
      {
        path: '/',
        name: 'Home',
        exact: true,
        component: loadable(() => import('../pages/Home')),
      },
      {
        path: '/about',
        name: 'About',
        exact: true,
        component: loadable(() => import('../pages/About')),
      },
      {
        path: '/blog',
        name: 'Blog',
        component: loadable(() => import('../pages/Blog')),
        fetchData: () => [PostService.getPosts()],
      },
      {
        path: '/blog/:id',
        name: 'Article',
        component: loadable(() => import('../pages/Article')),
      },
      {
        path: '/ui',
        name: 'Ui',
        component: loadable(() => import('../pages/Ui')),
      },
      {
        path: '/pwa',
        name: 'Pwa',
        component: loadable(() => import('../pages/Pwa')),
      },
      {
        path: '/login',
        name: 'Login',
      },
      {
        path: '/account',
        name: 'Account',
        component: loadable(() => import('../pages/Account')),
      },
      {
        path: '*',
        name: '404',
        component: loadable(() => import('../pages/NotFound')),
      },
    ],
  },
];
