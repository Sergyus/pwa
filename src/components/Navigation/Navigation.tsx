import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from '@material-ui/core';
import './navigation.scss';
import { useTranslation } from 'react-i18next';
import { headerMenu } from '@components/Navigation/consts';

export default function Navigation(): JSX.Element {
  const { t } = useTranslation('menu');

  return (
    <div data-testid="menu" className="nav-header">
      {headerMenu.map((item) => (
        <Button key={item.name} color="inherit">
          <Link
            style={{ color: '#fff', textDecoration: 'none' }}
            to={item.link}
            data-testid="menuItem"
          >
            {t(item.name)}
          </Link>
        </Button>
      ))}
    </div>
  );
}
