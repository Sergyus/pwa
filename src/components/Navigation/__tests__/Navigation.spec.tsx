import React from 'react';
import { render, screen } from '@testing-library/react';
import Navigation from '@components/Navigation';
import { MemoryRouter } from 'react-router-dom';
import { headerMenu } from '@components/Navigation/consts';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('<Navigation/>', () => {
  it('should renders', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const menu = screen.getByTestId('menu');
    expect(menu).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render all items', () => {
    render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const menuItems = screen.getAllByTestId('menuItem');
    expect(menuItems).toHaveLength(headerMenu.length);
  });
});
