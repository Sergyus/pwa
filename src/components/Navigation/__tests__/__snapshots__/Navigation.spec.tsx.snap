// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> should renders 1`] = `
<DocumentFragment>
  <div
    class="nav-header"
    data-testid="menu"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a
          data-testid="menuItem"
          href="/"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          home
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a
          data-testid="menuItem"
          href="/blog"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          blog
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a
          data-testid="menuItem"
          href="/about"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          about
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a
          data-testid="menuItem"
          href="/ui"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          Ui
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a
          data-testid="menuItem"
          href="/pwa"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          Pwa
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a
          data-testid="menuItem"
          href="/apollo"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          Apollo
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
